public/app/features/dashboard/services/PublicDashboardDataSource.ts

```diff
@@ -78,23 +88,51 @@ export class PublicDashboardDataSource extends DataSourceApi<DataQuery, DataSour
       return of({ data: [] });
     }
 
-    const body: any = { intervalMs, maxDataPoints };
-
-    return getBackendSrv()
-      .fetch<BackendDataSourceResponse>({
-        url: `/api/public/dashboards/${publicDashboardAccessToken}/panels/${panelId}/query`,
-        method: 'POST',
-        data: body,
-        requestId,
-      })
-      .pipe(
-        switchMap((raw) => {
-          return of(toDataQueryResponse(raw, queries));
-        }),
-        catchError((err) => {
-          return of(toDataQueryResponse(err));
+    // Its an annotations query
+    // Currently, annotations requests come in one at a time, so there will only be one target
+    const target = request.targets[0];
+    if (target.queryType === GrafanaQueryType.Annotations) {
+      if (target?.datasource?.uid === GRAFANA_DATASOURCE_NAME) {
+        return from(this.getAnnotations(request));
+      }
+      return of({ data: [] });
+    }
+
+    // Its a datasource query
+    else {
+      const body: any = { intervalMs, maxDataPoints };
+
+      return getBackendSrv()
+        .fetch<BackendDataSourceResponse>({
+          url: `/api/public/dashboards/${publicDashboardAccessToken}/panels/${panelId}/query`,
+          method: 'POST',
+          data: body,
+          requestId,
         })
-      );
+        .pipe(
+          switchMap((raw) => {
+            return of(toDataQueryResponse(raw, queries));
+          }),
+          catchError((err) => {
+            return of(toDataQueryResponse(err));
+          })
+        );
+    }
+  }
+
+  async getAnnotations(request: DataQueryRequest<DataQuery>): Promise<DataQueryResponse> {
+    const {
+      publicDashboardAccessToken: accessToken,
+      range: { to, from },
+    } = request;
+
+    const params = {
+      from: from.valueOf(),
+      to: to.valueOf(),
+    };
+    const annotations = await getBackendSrv().get(`/api/public/dashboards/${accessToken}/annotations`, params);
+
+    return { data: [toDataFrame(annotations)] };
```

Comment:
Question: would it be a smart idea to return it as data frames directly from the backend instead of needing to transform the response in the frontend? Or do we have any other consumers of this API a part of this frontend?